distributed/actor.py

```diff
@@ -227,34 +253,61 @@ class ActorFuture:
     Actor
     """
 
-    def __init__(self, q, io_loop, result=None):
-        self.q = q
-        self.io_loop = io_loop
-        if result:
-            self._cached_result = result
-        self.status = "pending"
+    @abc.abstractmethod
+    def __await__(self):
+        pass
+
+    @abc.abstractmethod
+    def result(self, timeout=None):
+        pass
+
+    @abc.abstractmethod
+    def done(self):
+        pass
+
+    def __repr__(self):
+        return "<ActorFuture>"
+
+
+class _EagerActorFuture(ActorFuture):
+    """Future to an actor's method call when an actor calls another actor on the same worker"""
+
+    def __init__(self, result):
+        self._result = result
+
+    def __await__(self):
+        return self._result
+        yield
```

Comment:
Is this `yield` serving any purpose? Doesn't seem so to me.

Reply:
`__await__` needs to return a generator. The unreachable `yield` makes this function one:

```python
import asyncio

class _EagerActorFuture:
    def __init__(self, v):
        self._v = v

    def __await__(self):
        return self._v

async def foo():
    return await _EagerActorFuture(1)


assert asyncio.run(foo()) == 1
```

an alternative is something like this:

```python
def __await__(self):
    if False:
        yield
    return self._result
 ```

Reply:
your comment made me go back and remove the `yield` and all the tests passed - because I was still not covering this case!

I can't think of a better way to cover this, as it seems a very narrow corner case:
https://github.com/dask/distributed/blob/1a09c7e606ee7c9dcc9e4d778b68dd04d5167072/distributed/tests/test_actor.py#L640-L655

Reply:
ah `get_client().sync(ac.increment)` is a realistic usecase of `_EagerActorFuture.__await__`

Reply:
This seems worth a comment in the code. Also, another thing that type annotations would make clearer. If you annotated the return type on the `ActorFuture` ABC as `Awaitable[T]` (or maybe `Coroutine[T, None, None] | Coroutine[None, None, T]` is necessary? Not sure.), then
1. mypy should hopefully error if you leave out the `yield` here?
2. It would make it make it more strictly obvious and documentable that `__await__` needs to be a generator.

Reply:
yep - `Awaitable[_T]` forces `__await__` to return a `Generator[object, None, _T]`